pymc_marketing/mmm/budget_optimizer.py

```diff
@@ -0,0 +1,79 @@
+# optimization_utils.py
+from typing import List
+
+import numpy as np
+from pandas import DataFrame
+from xarray import DataArray
+
+from pymc_marketing.mmm.utils import CurveCalculator
+
+
+def cost_function(
```

Comment:
As we are using the class `CurveCalculator` to have a polynomial fit. Can we add a more descriptive name to this method? Something like `compute_total_cost_after_budget_optimization` or something like that (be creative ;) )

Reply:
also, can we add unit-tests for this function as well :)